beacon-chain/core/blocks/block_operations.go

```diff
@@ -830,6 +870,9 @@ func ProcessDeposits(
 	var err error
 	deposits := body.Deposits
 	for _, deposit := range deposits {
+		if deposit == nil || deposit.Data == nil {
+			return nil, fmt.Errorf("nil deposit in block body: %v", body)
```

Comment:
shouldn't we just continue here?

Reply:
not sure about it
why should ssz deserialise deposits with a nil entry?
wdyt?

Reply:
so a bad deposit can fail a whole block? i always thought deposit is an independent operation

Reply:
ok,
changing to continue